### PR TITLE
Move everything into an Envoy namespace in advance of same change in envoy repo

### PR DESCRIPTION
**Pull request by tschroed**

### CODE REVIEW

File: echo2.cc [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 #include "common/common/assert.h"
 
+namespace Lyft {
```
Comment (htuch) [inline echo2.cc]: Do we want to commit to Lyft as the namespace if one day this becomes CNCF? Maybe it should be Envoy? @mattklein123

Comment (mattklein123) [inline echo2.cc]: So what is the plan here? Are we somehow going to automate putting all code in the repo in a namespace? Yeah I agree putting it in Envoy namespace seems better.

**Comment by tschroed** [comment]
https://github.com/tschroed/envoy/commit/a0a5b3a25889694f845254f4fe81db463b7fa7f3

File: echo2.cc (hunk 1 above)
Comment (tschroed) [inline echo2.cc]: Bah, previous response ended up as an orphaned comment.

I can put it in Envoy if you prefer, but that seems like it's sufficiently generic that there might be other collisions whereas Lyft is unlikely to be grabbed by someone else.

Comment (mattklein123) [inline echo2.cc]: I don't really have a strong opinion here. Another option I guess is "EnvoyProxy". Very unlikely to conflict, but longer. Are there any other potential situations we could use an an example? @caniszczyk can you advise?

Comment (tschroed) [inline echo2.cc]: @htuch has convinced me that we should use Envoy.

File: WORKSPACE [hunk 2]
```diff
@@ -2,11 +2,22 @@ workspace(name = "envoy")
 
 local_repository(
     name = "envoy",
-    path = "envoy",
+    path = "/source",
```
Comment (htuch) [inline WORKSPACE]: I don't think you want your WORKSPACE in this PR.

File: bazel/get_workspace_status [hunk 3]
```diff
@@ -1 +1 @@
-../envoy/bazel/get_workspace_status
\ No newline at end of file
+/source/bazel/get_workspace_status
```
Comment (htuch) [inline bazel/get_workspace_status]: Ditto etc.

File: WORKSPACE [hunk 4]
```diff
@@ -5,8 +5,19 @@ local_repository(
...
+    path = "@envoy//ci/prebuilt",
+    skip_protobuf_bzl = True,
+)
+
+new_local_repository(
+    name = "protobuf_bzl",
+    path = "/thirdparty/protobuf",
+    # We only want protobuf.bzl, so don't support building out of this repo.
+    build_file_content = "",
+)
```
Comment (htuch) [inline WORKSPACE]: Nup, try again :)

Comment (tschroed) [inline WORKSPACE]: Git hates me. The feeling is mutual.

**Pull request merged**